addons/io/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/accessories/HomekitThermostatImpl.java

```diff
@@ -0,0 +1,287 @@
+                && currentTemperatureItemName != null && heatingCoolingModeItemName != null;
+    }
+
+    @Override
+    public CompletableFuture<Double> getCoolingThresholdTemperature() {
+        Item item = getItemRegistry().get(coolingThresholdItemName);
+        DecimalType state = (DecimalType) item.getStateAs(DecimalType.class);
+        if (state == null) {
+            return CompletableFuture.completedFuture(null);
+        }
+        return CompletableFuture.completedFuture(convertToCelsius(state.doubleValue()));
+    }
+
+    @Override
+    public CompletableFuture<ThermostatMode> getCurrentMode() {
+        Item item = getItemRegistry().get(heatingCoolingModeItemName);
+        StringType state = (StringType) item.getStateAs(StringType.class);
+        ThermostatMode mode;
+        if (state != null) {
+            String stringValue = state.toString();
+
+            if (stringValue.equals(settings.getThermostatCoolMode())) {
+                mode = ThermostatMode.COOL;
+            } else if (stringValue.equals(settings.getThermostatHeatMode())) {
+                mode = ThermostatMode.HEAT;
+            } else if (stringValue.equals(settings.getThermostatAutoMode())) {
+                mode = ThermostatMode.AUTO;
+            } else if (stringValue.equals(settings.getThermostatOffMode())) {
+                mode = ThermostatMode.OFF;
+            } else {
+                logger.error("Unrecognized heating cooling target mode: " + stringValue
+                        + ". Expected cool, heat, auto, or off strings in value.");
+                mode = ThermostatMode.OFF;
+            }
+        } else {
+            logger.info("Heating cooling target mode not available. Relaying value of OFF to Homekit");
+            mode = ThermostatMode.OFF;
+        }
+        return CompletableFuture.completedFuture(mode);
+    }
+
+    @Override
+    public CompletableFuture<Double> getCurrentTemperature() {
+        Item item = getItemRegistry().get(currentTemperatureItemName);
+        DecimalType state = (DecimalType) item.getStateAs(DecimalType.class);
+        if (state == null) {
+            return CompletableFuture.completedFuture(null);
+        }
+        return CompletableFuture.completedFuture(convertToCelsius(state.doubleValue()));
+    }
+
+    @Override
+    public CompletableFuture<Double> getHeatingThresholdTemperature() {
+        Item item = getItemRegistry().get(heatingThresholdItemName);
+        DecimalType state = (DecimalType) item.getStateAs(DecimalType.class);
+        if (state == null) {
+            return CompletableFuture.completedFuture(null);
+        }
+        return CompletableFuture.completedFuture(convertToCelsius(state.doubleValue()));
+    }
+
+    @Override
+    public CompletableFuture<ThermostatMode> getTargetMode() {
+        return getCurrentMode();
+    }
+
+    @Override
+    public CompletableFuture<Double> getTargetTemperature() {
+        if (autoThresholdItemName != null) {
```

Comment:
Alright, I spent some time tonight experimenting with Siri and thermostats, and discovered something interesting. Siri doesn't yet work with thermostats that expose separate heating and cooling targets! Although some apps do support it, I think it's probably an edge case for now. I've refactored the accessory to only use a single targetTemperature.